--- origsrc/libtorrent-rasterbar-0.14.12/include/libtorrent/config.hpp	2010-03-03 12:17:55.000000000 -0600
+++ src/libtorrent-rasterbar-0.14.12/include/libtorrent/config.hpp	2011-01-11 00:06:20.551003100 -0600
@@ -75,6 +75,8 @@ POSSIBILITY OF SUCH DAMAGE.
 	|| defined __OpenBSD__ || defined __bsdi__ || defined __DragonFly__ \
 	|| defined __FreeBSD_kernel__
 #define TORRENT_BSD
+#elif defined __CYGWIN__
+#define TORRENT_CYGWIN
 #elif defined __linux__
 #define TORRENT_LINUX
 #elif defined __MINGW32__
@@ -90,7 +92,7 @@ POSSIBILITY OF SUCH DAMAGE.
 #define TORRENT_BSD
 #endif
 
-#if defined TORRENT_BSD || defined TORRENT_LINUX || defined TORRENT_SOLARIS
+#if defined TORRENT_BSD || defined TORRENT_CYGWIN || defined TORRENT_LINUX || defined TORRENT_SOLARIS
 #define TORRENT_USE_RLIMIT 1
 #else
 #define TORRENT_USE_RLIMIT 0
--- origsrc/libtorrent-rasterbar-0.14.12/src/enum_net.cpp	2009-07-02 01:39:03.000000000 -0500
+++ src/libtorrent-rasterbar-0.14.12/src/enum_net.cpp	2011-01-11 00:28:09.311683500 -0600
@@ -41,12 +41,14 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <boost/asio/ip/host_name.hpp>
 #endif
 
-#if defined TORRENT_BSD || defined TORRENT_SOLARIS
+#if defined TORRENT_BSD || defined TORRENT_CYGWIN || defined TORRENT_SOLARIS
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <net/if.h>
+#if !defined TORRENT_CYGWIN
 #include <net/route.h>
+#endif
 #include <string.h>
 #include <boost/scoped_array.hpp>
 #endif
@@ -204,7 +206,7 @@ namespace libtorrent { namespace
 #endif
 
 
-#ifdef TORRENT_BSD
+#if defined TORRENT_BSD
 	bool verify_sockaddr(sockaddr_in* sin)
 	{
 		return (sin->sin_len == sizeof(sockaddr_in)
@@ -248,7 +250,7 @@ namespace libtorrent
 	{
 		std::vector<ip_interface> ret;
 // covers linux, MacOS X and BSD distributions
-#if defined TORRENT_LINUX || defined TORRENT_BSD || defined TORRENT_SOLARIS
+#if defined TORRENT_LINUX || defined TORRENT_BSD || defined TORRENT_CYGWIN || defined TORRENT_SOLARIS
 		int s = socket(AF_INET, SOCK_DGRAM, 0);
 		if (s < 0)
 		{
@@ -304,7 +306,7 @@ namespace libtorrent
 
 #if defined TORRENT_BSD
 			int current_size = item.ifr_addr.sa_len + IFNAMSIZ;
-#elif defined TORRENT_LINUX || defined TORRENT_SOLARIS
+#elif defined TORRENT_LINUX || defined TORRENT_SOLARIS || defined TORRENT_CYGWIN
 			int current_size = sizeof(ifreq);
 #endif
 			ifr += current_size;
