--- origsrc/libtorrent-rasterbar-0.14.12/src/Makefile.am	2009-09-12 17:00:51.000000000 -0500
+++ src/libtorrent-rasterbar-0.14.12/src/Makefile.am	2011-01-10 21:18:50.326327700 -0600
@@ -108,7 +108,7 @@ $(top_srcdir)/include/libtorrent/variant
 $(top_srcdir)/include/libtorrent/version.hpp
 
 
-libtorrent_rasterbar_la_LDFLAGS = $(LDFLAGS) -version-info $(INTERFACE_VERSION_INFO)
+libtorrent_rasterbar_la_LDFLAGS = $(LDFLAGS) -version-info $(INTERFACE_VERSION_INFO) -no-undefined
 libtorrent_rasterbar_la_LIBADD = @ZLIB@ @BOOST_SYSTEM_LIB@ @BOOST_FILESYSTEM_LIB@ @BOOST_THREAD_LIB@ @PTHREAD_LIBS@ @SSL_LIBS@
 
 AM_CXXFLAGS= -ftemplate-depth-100 -I$(top_srcdir)/include -I$(top_srcdir)/include/libtorrent @ZLIBINCL@ @DEBUGFLAGS@ @PTHREAD_CFLAGS@ -DBOOST_MULTI_INDEX_DISABLE_SERIALIZATION @SSL_CXXFLAGS@
