--- origsrc/libtorrent-rasterbar-0.15.6/include/libtorrent/config.hpp	2011-02-21 21:22:38.000000000 -0600
+++ src/libtorrent-rasterbar-0.15.6/include/libtorrent/config.hpp	2011-06-10 16:07:33.874314000 -0500
@@ -125,6 +125,8 @@ POSSIBILITY OF SUCH DAMAGE.
 	|| defined __FreeBSD_kernel__
 #define TORRENT_BSD
 #define TORRENT_HAS_FALLOCATE 0
+#elif defined __CYGWIN__
+#define TORRENT_CYGWIN
 #elif defined __linux__
 #define TORRENT_LINUX
 #elif defined __MINGW32__
@@ -148,7 +150,7 @@ POSSIBILITY OF SUCH DAMAGE.
 #define TORRENT_USE_WRITEV 1
 #define TORRENT_USE_IOSTREAM 1
 
-#if defined TORRENT_BSD || defined TORRENT_LINUX || defined TORRENT_SOLARIS
+#if defined TORRENT_BSD || defined TORRENT_CYGWIN || defined TORRENT_LINUX || defined TORRENT_SOLARIS
 #define TORRENT_USE_RLIMIT 1
 #else
 #define TORRENT_USE_RLIMIT 0
--- origsrc/libtorrent-rasterbar-0.15.6/include/libtorrent/error_code.hpp	2010-03-02 03:01:39.000000000 -0600
+++ src/libtorrent-rasterbar-0.15.6/include/libtorrent/error_code.hpp	2011-06-10 16:34:41.978412300 -0500
@@ -36,7 +36,7 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <boost/version.hpp>
 #include <boost/shared_ptr.hpp>
 
-#if defined TORRENT_WINDOWS || defined TORRENT_CYGWIN
+#if defined TORRENT_WINDOWS
 // asio assumes that the windows error codes are defined already
 #include <winsock2.h>
 #endif
--- origsrc/libtorrent-rasterbar-0.15.6/src/broadcast_socket.cpp	2011-02-16 23:03:24.000000000 -0600
+++ src/libtorrent-rasterbar-0.15.6/src/broadcast_socket.cpp	2011-06-10 16:33:05.477044900 -0500
@@ -38,7 +38,7 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #include "libtorrent/config.hpp"
 
-#if defined TORRENT_WINDOWS || defined TORRENT_CYGWIN
+#if defined TORRENT_WINDOWS
 // asio assumes that the windows error codes are defined already
 #include <winsock2.h>
 #endif
--- origsrc/libtorrent-rasterbar-0.15.6/src/enum_net.cpp	2011-02-15 00:34:34.000000000 -0600
+++ src/libtorrent-rasterbar-0.15.6/src/enum_net.cpp	2011-06-10 16:13:05.472491300 -0500
@@ -41,12 +41,14 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <boost/asio/ip/host_name.hpp>
 #endif
 
-#if defined TORRENT_BSD || defined TORRENT_SOLARIS
+#if defined TORRENT_BSD || defined TORRENT_CYGWIN || defined TORRENT_SOLARIS
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <net/if.h>
+#if !defined TORRENT_CYGWIN
 #include <net/route.h>
+#endif
 #include <string.h>
 #include <boost/scoped_array.hpp>
 #endif
@@ -315,7 +317,7 @@ namespace libtorrent
 		}
 		freeifaddrs(ifaddr);
 // MacOS X, BSD and solaris
-#elif defined TORRENT_BSD || defined TORRENT_SOLARIS
+#elif defined TORRENT_BSD || defined TORRENT_CYGWIN || defined TORRENT_SOLARIS
 		int s = socket(AF_INET, SOCK_DGRAM, 0);
 		if (s < 0)
 		{
@@ -374,7 +376,7 @@ namespace libtorrent
 
 #if defined TORRENT_BSD
 			int current_size = item.ifr_addr.sa_len + IFNAMSIZ;
-#elif defined TORRENT_LINUX || defined TORRENT_SOLARIS
+#elif defined TORRENT_CYGWIN || defined TORRENT_LINUX || defined TORRENT_SOLARIS
 			int current_size = sizeof(ifreq);
 #endif
 			ifr += current_size;
--- origsrc/libtorrent-rasterbar-0.15.6/src/natpmp.cpp	2011-02-20 23:25:59.000000000 -0600
+++ src/libtorrent-rasterbar-0.15.6/src/natpmp.cpp	2011-06-10 16:33:56.069532900 -0500
@@ -37,7 +37,7 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #include "libtorrent/config.hpp"
 
-#if defined TORRENT_WINDOWS || defined TORRENT_CYGWIN
+#if defined TORRENT_WINDOWS
 // asio assumes that the windows error codes are defined already
 #include <winsock2.h>
 #endif
